### PR TITLE
feat: multi-did creation implementation for an organization

### PR DESCRIPTION
**Pull request by tipusinghaw**
#### What ####
- Implemented set primary DID while creating the DID
- Implemented fetch all DID for an organization
- Implemented set (Update) primary DID for an organization
#### Why ####
- Organization can able to hold multiple DIDs

### CODE REVIEW

File: apps/agent-service/src/interface/agent-service.interface.ts [hunk 1]
```diff
@@ -183,6 +184,15 @@ export interface IStoreOrgAgentDetails {
   agentType?: string;
 }
 
+export interface StoreDidDetails {
```
Comment (bhavanakarwade) [inline apps/agent-service/src/interface/agent-service.interface.ts:187]: We are following the prefix 'I' before the interface name

**Review by bhavanakarwade — CHANGES_REQUESTED** [review]

File: apps/agent-service/src/repositories/agent-service.repository.ts [hunk 2]
```diff
@@ -155,6 +155,58 @@ export class AgentServiceRepository {
...
+
+    /**
+     * Set primary DID
+     * @param did
+     * @returns did details
+     */
+    // eslint-disable-next-line camelcase
+    async setPrimaryDid(isPrimaryDid:string, orgId:string): Promise<org_agents> {
+        try {
+          return await this.prisma.org_agents.update({
```
Comment (KulkarniShashank) [inline apps/agent-service/src/repositories/agent-service.repository.ts:194]: Remove unnecessary 'await' in function call
```suggestion
          return this.prisma.org_agents.update({
```

File: apps/api-gateway/src/organization/organization.controller.ts [hunk 3]
```diff
@@ -318,6 +319,29 @@ export class OrganizationController {
...
+   /**
+  * @returns DID list of organization
+  */
+
+   @Get('/:orgId/dids')
+   @Roles(OrgRoles.OWNER, OrgRoles.ADMIN, OrgRoles.ISSUER)
+   @ApiBearerAuth()
+   @UseGuards(AuthGuard('jwt'), OrgRolesGuard)
+   @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: ApiResponseDto })
+   @ApiOperation({ summary: 'Fetch organization dids', description: 'Get all dids from organization' })
```
Comment (KulkarniShashank) [inline apps/api-gateway/src/organization/organization.controller.ts:332]: dids must be `DIDs` in a summary and description
```suggestion
   @ApiOperation({ summary: 'Fetch organization DIDs', description: 'Get all DIDs from organization' })
```

File: apps/agent-service/src/interface/agent-service.interface.ts (hunk 1 above)
Comment (KulkarniShashank) [reply apps/agent-service/src/interface/agent-service.interface.ts:187]:
```suggestion
export interface IStoreDidDetails {
```

File: apps/organization/interfaces/organization.interface.ts [hunk 4]
```diff
@@ -141,4 +143,30 @@ export interface Payload {
   pageNumber: number;
   pageSize: number;
   search: string;
+}
+
+export interface DidList {
```
Comment (KulkarniShashank) [inline apps/organization/interfaces/organization.interface.ts:148]:
```suggestion
export interface IDidList {
```

File: apps/organization/interfaces/organization.interface.ts [hunk 5]
```diff
@@ -141,4 +143,30 @@ export interface Payload {
...
+  id: string;
+  createDateTime: Date;
+  did: string;
+  lastChangedDateTime: Date;
+  isPrimaryDid: boolean;
+}
+
+export interface PrimaryDid {
```
Comment (KulkarniShashank) [inline apps/organization/interfaces/organization.interface.ts:156]:
```suggestion
export interface IPrimaryDid {
```

File: apps/organization/interfaces/organization.interface.ts [hunk 6]
```diff
@@ -141,4 +143,30 @@ export interface Payload {
...
+  orgId: string,
+  did: string
+}
+
+export interface DidDetails {
```
[the remark this answers is not in the document]
Comment (KulkarniShashank) [reply apps/organization/interfaces/organization.interface.ts:161]:
```suggestion
export interface IDidDetails {
```

File: apps/organization/repositories/organization.repository.ts [hunk 7]
```diff
@@ -734,4 +734,84 @@ export class OrganizationRepository {
...
+          id: true,
+          createDateTime: true,
+          did: true,
+          lastChangedDateTime: true,
+          isPrimaryDid: true  
+        }
+      });
+    } catch (error) {
+      this.logger.error(`error in getAllOrganizationDid: ${JSON.stringify(error)}`);
+      throw new InternalServerErrorException(error);
```
Comment (KulkarniShashank) [inline apps/organization/repositories/organization.repository.ts:754]: "Updated error handling to ensure compatibility with RpcException"
```suggestion
      throw error;
```

File: libs/common/src/response-messages/index.ts [hunk 8]
```diff
@@ -111,7 +113,9 @@ export const ResponseMessages = {
             ecosystemIdIsRequired:'ecosystemId is required',
             roleNotMatch: 'User does not have access',
             orgDoesNotMatch: 'Organization does not match',
-            invalidClient: 'Invalid client credentials'
+            invalidClient: 'Invalid client credentials',
+            primaryDid: 'This did is already set to primary DID',
```
Comment (KulkarniShashank) [inline libs/common/src/response-messages/index.ts:117]:
```suggestion
            primaryDid: 'This DID is already set to primary DID',
```

File: libs/common/src/response-messages/index.ts [hunk 9]
```diff
@@ -111,7 +113,9 @@ export const ResponseMessages = {
...
+            didNotFound: 'Did does not exist in organiation'
```
Comment (KulkarniShashank) [inline libs/common/src/response-messages/index.ts:118]:
```suggestion
            didNotFound: 'DID does not exist in organiation'
```

**Review by bhavanakarwade — APPROVED** [review]
LG

**Review by KulkarniShashank — APPROVED** [review]
LGTM

**Pull request merged**